kubernetes/resource_kubernetes_secret_v1_data.go

```diff
@@ -0,0 +1,249 @@
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+)
+
+func resourceKubernetesSecretV1Data() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceKubernetesSecretV1DataCreate,
+		ReadContext:   resourceKubernetesSecretV1DataRead,
+		UpdateContext: resourceKubernetesSecretV1DataUpdate,
+		DeleteContext: resourceKubernetesSecretV1DataDelete,
+
+		Schema: map[string]*schema.Schema{
+			"metadata": {
+				Type:        schema.TypeList,
+				Description: "Metadata for the kubernetes Secret.",
+				Required:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:        schema.TypeString,
+							Description: "The name of the Secret.",
+							Required:    true,
+							ForceNew:    true,
+						},
+						"namespace": {
+							Type:        schema.TypeString,
+							Description: "The namespace of the Secret.",
+							Optional:    true,
+							ForceNew:    true,
+							Default:     "default",
+						},
+					},
+				},
+			},
+			"data": {
+				Type:        schema.TypeMap,
+				Description: "Data to be stored in the Kubernetes Secret.",
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"force": {
+				Type:        schema.TypeBool,
+				Description: "Flag to force updates to the Kubernetes Secret.",
+				Optional:    true,
+			},
+			"field_manager": {
+				Type:        schema.TypeString,
+				Description: "Set the name of the field manager for the specified labels",
+				Optional:    true,
+				Default:     defaultFieldManagerName,
+			},
+		},
+	}
+}
+
+func resourceKubernetesSecretV1DataCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	metadata := expandMetadata(d.Get("metadata").([]interface{}))
+	// Sets the resource id based on the metadata
+	d.SetId(buildId(metadata))
+
+	//Calling the update function ensuring resource config is correct
+	diag := resourceKubernetesSecretV1DataUpdate(ctx, d, m)
+	if diag.HasError() {
+		d.SetId("")
+	}
+	return diag
+}
+
+// Retrieves the current state of the k8s secret, and update the current sate
+func resourceKubernetesSecretV1DataRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	conn, err := m.(KubeClientsets).MainClientset()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	namespace, name, err := idParts(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	// getting the secret data
+	res, err := conn.CoreV1().Secrets(namespace).Get(ctx, name, v1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return diag.Diagnostics{{
+				Severity: diag.Warning,
+				Summary:  "Secret deleted",
+				Detail:   fmt.Sprintf("The underlying secret %q has been deleted. You should recreate the underlying secret, or remove it from your configuration.", name),
+			}}
+		}
+		return diag.FromErr(err)
+	}
+
+	configuredData := d.Get("data").(map[string]interface{})
```

Comment:
I know you've just copied this from code elsewhere in the provider, but in new code we should probably use `any` instead of `interface{}`.

Reply:
@jrhouston  Do we want to introduce the use of `any`? [I know that it's just an alias for `interface{}` ](https://go.dev/blog/intro-generics)which would make it easier to read but if we start introducing this we'll also want to apply this across the whole repo so that it's consistent.

Reply:
I'm OK with any new code being introduced being `any`, we don't have to go back and change it everywhere in the old code.